pkg/cloud/oracle/usageapiintegration.go

```diff
@@ -0,0 +1,107 @@
+package oracle
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/opencost/opencost/core/pkg/opencost"
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/example/helpers"
+	"github.com/oracle/oci-go-sdk/v65/usageapi"
+)
+
+type UsageApiIntegration struct {
+	UsageApiConfiguration
+}
+
+func (uai *UsageApiIntegration) GetCloudCost(start time.Time, end time.Time) (*opencost.CloudCostSetRange, error) {
+	client, err := uai.GetUsageApiClient()
+	if err != nil {
+		return nil, fmt.Errorf("getting oracle usage api client: %s", err.Error())
+	}
+
+	req := usageapi.RequestSummarizedUsagesRequest{
+		RequestSummarizedUsagesDetails: usageapi.RequestSummarizedUsagesDetails{
+			CompartmentDepth: common.Float32(5.749769), //todo: what does this mean
+			Granularity:      usageapi.RequestSummarizedUsagesDetailsGranularityDaily,
+			GroupBy:          []string{"group-by"},
+			GroupByTag: []usageapi.Tag{
+				{
+					Key:       common.String("key"),
+					Namespace: common.String("namespace"),
+					Value:     common.String("value"),
+				},
+			},
+			IsAggregateByTime: common.Bool(false),
+			TimeUsageStarted:  &common.SDKTime{Time: start},
+			TimeUsageEnded:    &common.SDKTime{Time: end},
+			Filter: &usageapi.Filter{
+				Dimensions: []usageapi.Dimension{
+					{
+						Key:   common.String("key"),
+						Value: common.String("value"),
+					},
+				},
+				Operator: usageapi.FilterOperatorAnd,
+				Tags: []usageapi.Tag{
+					{
+						Key:       common.String("key"),
+						Namespace: common.String("namespace"),
+						Value:     common.String("value"),
+					},
+				},
+			},
+			QueryType: usageapi.RequestSummarizedUsagesDetailsQueryTypeCost,
+			TenantId:  common.String("tenantid"),
+		},
+		Limit: common.Int(533),
+	}
+
+	resp, err := client.RequestSummarizedUsages(context.Background(), req)
+	helpers.FatalIfError(err)
+
+	ccsr := &opencost.CloudCostSetRange{}
+
+	for _, item := range resp.Items {
+		properties := &opencost.CloudCostProperties{
+			ProviderID:      opencost.OracleProvider,
+			Provider:        opencost.OracleProvider,
+			AccountID:       *item.TenantId,
+			AccountName:     *item.TenantName,
+			InvoiceEntityID: *item.SubscriptionId,
+			RegionID:        *item.Region,
+			Service:         *item.Service,
+			Category:        *item.Platform, //todo: is this accurate to Category?
```

Comment:
Category is generally a derived field for us, this will be its own function based off of service and possibly a result description that wont be included as is